pkg/version/version_info.go

```diff
@@ -0,0 +1,68 @@
+package version
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+)
+
+type versionInfo struct {
+	major   int
+	minor   int
+	release int
+}
+
+// compareVersionInfo returns:
+// 	0: if a == b
+//  n > 0: if a > b
+//  n < 0: if a < b
+//  0 with error: if a == nil || b == nil
+func compareVersionInfo(a *versionInfo, b *versionInfo) (int, error) {
+	if a == nil || b == nil {
+		return 0, fmt.Errorf("parameter must not be null")
+	}
+
+	// Check major version
+	result := a.major - b.major
+	if result != 0 {
+		return result, nil
+	}
+
+	// Major is equal, check minor
+	result = a.minor - b.minor
+	if result != 0 {
+		return result, nil
+	}
+
+	// Major and minor is equal, check release
+	result = a.release - b.release
+	return result, nil
+}
+
+func extractVersion(versionString string) (*versionInfo, error) {
+	mainVersionRegex := regexp.MustCompile(`[\d]+\.[\d]+\.[\d]+`)
+	mainVersion := mainVersionRegex.FindString(versionString)
+
+	versionMergeRegex := regexp.MustCompile(`[\d]+`)
```

Comment:
Golang supports groups/submatches which is what you're trying to get here: https://golang.org/pkg/regexp/#Regexp.FindStringSubmatch

You could also just split the string by dot.

Reply:
Changed to FindStringSubmatch.

Splitting the string by dots returns 3 values for e.g.: 1.203.0 but 4 values for e.g.: 1.203.0.20200915-101926. Opted for regex to keep that more consistent.
Furthermore the used regex checks if the values between dots are actually numbers which makes checking for a malformed version string easier